string node
float32 state
